File: bulk/go/csrf_01.go
// CWE-352: CSRF — state-changing endpoint with no token check
package main
import (
	"encoding/json"
	"net/http"
)
func transferHandler(w http.ResponseWriter, r *http.Request) {
	var req struct{ To string; Amount float64 }
	json.NewDecoder(r.Body).Decode(&req)
	// No CSRF token validation
	transfer(req.To, req.Amount)
	w.WriteHeader(http.StatusOK)
}
func transfer(string, float64) {}

File: bulk/javascript/csrf_01.js
// CWE-352: CSRF - no token validation on state-changing endpoint
const express = require('express');
const app = express();
app.post('/transfer', (req, res) => {
  const { to, amount } = req.body; // no CSRF token check
  transferFunds(to, amount);
  res.json({ ok: true });
});
function transferFunds() {}
